### ISSUE TITLE
const used in a for loop when targeting ES6 sails through redeclaration check and fails at runtime.

### ISSUE THREAD

**Issue by seanparsons** [Microsoft/TypeScript#18410]
**TypeScript Version:**  2.4.1

**Code**

```ts
function testThing(parameter: number) {
  for (const parameter of [1, 2, 3]) {
  }
}
testThing(5)
```

**Expected behavior:**

Should flag this up as a redeclaration as if a line like `const parameter = 1` had been duplicated.

**Actual behavior:**

In Safari at least this will blow up at runtime with `Cannot declare a const variable twice: parameter`.

**Comment by mhegazy** [comment]
Looks like a bug in JSC. Looking at the [evaluation semantics for `for` in the ES6 spec](https://www.ecma-international.org/ecma-262/6.0/#sec-for-statement-runtime-semantics-labelledevaluation), the spec is clear about how `const` vs. `let` are evaluated.  When evaluating ` ForBodyEvaluation` it only uses the `perIterationLets` which does not include the consts.

**Comment by RyanCavanaugh** [comment]
https://trac.webkit.org/changeset/217200/webkit/trunk/Source
